text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,1443 @@
+
+- Ease of use
+- Maintenance
+- Quality
+
+Feeding those signals are related measures of:
+
+- Popularity
+- Credibility
+
+We detail how we propose to address each of these in turn, plus a rating of the
+five crates from the user research survey as examples.
+
+We'd like to provide a coarse binning of the scores in each category, to avoid
+over-analyzing the difference between, say, 72% and 78% and seeing significance
+where there isn't really one. We've considered using letter grades, but those
+often have emotional associations (F means you're a failure), when it should be
+just an indicator of reality and not a value judgment. So we're also proposing
+an option of an emoji scale and are open to other proposals:
+
+<table>
+    <tr>
+        <th>Percentage</th>
+        <th>Letter grade</th>
+        <th>Emoji</th>
+    </tr>
+    <tr>
+        <td>>= 90%</td>
+        <td>A</td>
+        <td>☀️</td>
+    </tr>
+    <tr>
+        <td>80-89%</td>
+        <td>B</td>
+        <td>🌤</td>
+    </tr>
+    <tr>
+        <td>70-79%</td>
+        <td>C</td>
+        <td>⛅️</td>
+    </tr>
+    <tr>
+        <td>60-69%</td>
+        <td>D</td>
+        <td>🌥</td>
+    </tr>
+    <tr>
+        <td>50-59%</td>
+        <td>E</td>
+        <td>☁️</td>
+    </tr>
+    <tr>
+        <td>&lt;= 49%</td>
+        <td>F</td>
+        <td>🌧</td>
+    </tr>
+</table>
+
+### Ease of use
+
+By far, the most common attribute people said they considered in the survey was
+whether a crate had good documentation. Frequently mentioned when discussing
+documentation was the desire to quickly find an example of how to use the crate.
+
+- Percentage of top-level items that have documentation
+  - We have created a proof-of-concept [cargo doc-coverage][] tool to count the
+    number of public items and the percentage of those that have/don't have
+    documentation. The overall documentation coverage didn't match our human
+    perceptions of well-documentedness from looking at the front page of
+    documentation, so we decided top-level items are more important than items
+    in submodules. For example, nom is 48% documented overall, but the
+    top-level items are extremely well documented, 170/195 or 87%. Our
+    definition of "top-level" counts the overall crate as an item. We think our
+    doc coverage POC can be modified to report this number.
+  - Would need to unpack and run this on each package version in a background
+    job started by a publish; then save the percentage in crates.io's database.
+
+- In the crate root documentation, presence of a section headed with the word
+  "Example" and containing a codeblock
+  - Existing issue, seen in the survey results is that people look in both the
+    README of the repo and the front page of the docs for examples. We have an
+    opportunity to encourage at least one to be present reliably.
+  - Increases the doc percentage score by 5%
+
+- Presence of files in `/examples`
```

Comment:
`examples/` is already a standardized directory, as `cargo run --example <name>` runs `examples/<name>.rs`.